src/app/modules/datagrid/datagrid-controller.js

```diff
@@ -1,135 +1,252 @@
 (function () { // to wrap use strict
 	'use strict';
 
+	//This variable allows both the onExit function of the stateProvider and the controller to access the resizeGrid function
+	var resizeGrid;
+
+	var tagList = [];
+
 	// Model declaration for the data grid view (url #/datagrid)
-	angular.module('BloomLibraryApp.datagrid', ['ui.router', 'restangular', 'ngGrid'])//, 'BloomLibraryApp.detail'])
+	angular.module('BloomLibraryApp.datagrid', ['ui.router', 'restangular', 'ui.grid', 'ui.grid.pagination', 'ui.grid.resizeColumns', 'ui.grid.edit', 'ui.grid.cellNav', 'ui.grid.autoResize', 'ngTagsInput'])//, 'BloomLibraryApp.detail'])
 	.config(['$stateProvider', function config($stateProvider) {
 
 		$stateProvider.state('datagrid', {
+			parent: 'requireLoginResolution',
 			url: "/datagrid",
 			templateUrl: 'modules/datagrid/datagrid.tpl.html',
-			controller: 'DataGridCtrl'
+			controller: 'DataGridCtrl',
+			title: 'Book Library',
+			onExit: function() {
+				window.removeEventListener("resize", resizeGrid);
+			}
 		});
 	} ]);
 
+	angular.module('BloomLibraryApp.datagrid')
+	.service('autoTags', ['tagService', '$q', function(tagService, $q) {
+		//var tags = tagService.getTags();
+
+		this.getTags = function(query) {
+			var deferred = $q.defer();
+			var matches = [];
+			//Match beginning
+			var firstRegex = new RegExp('^' + query, 'i');
+			//Match word starts
+			var wordRegex = new RegExp('\\W' + query, 'i');
+			//Match camel case
+			var capQuery = query.charAt(0).toUpperCase() + query.substr(1, query.length - 1);
+			var camelRegex = new RegExp('[a-z]' + capQuery);
+
+			//This should be replaced by the version below
+			for(var i in tagList) {
+				if(firstRegex.test(i) || wordRegex.test(i) || camelRegex.test(i)) {
+					matches.push({"text": i});
+				}
+			}
+
+			//This is the real version to use when the tagService is straightened out
```

Comment:
Can you say a little more? What needs to be straightened out and why haven't you?